workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs

```diff
@@ -801,37 +758,11 @@ private BuildOptions ConfigureAndroid(BuildTargetConfig buildTarget)
                 options &= ~BuildOptions.Development;
```

Comment:
Currently you can get into a state where development is off and debug is on:

- Enabled Development
- Enable Debug
- Disable Development

The UI will show them both disabled, but the build options will still have the debug flag set.

Should this line also turn off the debug flag for consistency between the UI & the backing data?

Reply:
I agree that it's a good idea to explicitly disabled the flag when the `development` toggle is disabled.

Something interesting that I didn't realise happened in the UI code:

Starting at the third step in your comment
- Development flag is disabled
- The UI and flags are now in an inconsistent state since the UI shows the `Allow Debug` option as disabled whereas the flag internally is enabled
- On the next render of the UI, in the `ConfigureDebug` function the `isDebugEnabled` variable is now `false`. This means the if statement inside the `using` block will always trigger the `else` case causing the `AllowDebugging` flag to be disabled.

Even though I don't explicitly disable the flag, due to the way `EditorGUILayout.Toggle()` works in the `EditorGUI.DisabledScope` it will always return false when the development flag is disabled causing an update to the `AllowDebugging` flag and disabling the flag anyway.